lib/ret_web/controllers/page_controller.ex

```diff
@@ -214,6 +219,24 @@ defmodule RetWeb.PageController do
     |> send_resp(200, chunks)
   end
 
+  defp cors_proxy(%Conn{request_path: "/" <> url, query_string: ""} = conn), do: cors_proxy(conn, url)
+  defp cors_proxy(%Conn{request_path: "/" <> url, query_string: qs} = conn), do: cors_proxy(conn, "#{url}?#{qs}")
+
+  defp cors_proxy(conn, url) do
+    allowed_origins = Application.get_env(:ret, RetWeb.Endpoint)[:allowed_origins] |> String.split(",")
+    opts = ReverseProxyPlug.init(upstream: url, allowed_origins: allowed_origins, proxy_url: RetWeb.Endpoint.url())
```

Comment:
note we can't use this plug as a plug because the upstream URL is completely dynamic, not fixed. an alternative approach to this would be to just not use this plug library and lift most of the code into reticulum proper, but most of the code in there can be re-used and so it seemed best to work against a fork of it.